enhancements/insights/insights-runtime-extractor.md

```diff
@@ -0,0 +1,647 @@
+
+###### Risk of misbehaving code execution
+
+A few fingerprints relies on executing process in the user namespace (for example, executing `node --version` to determine the version of Node.js).
+There is a risk of side-effects if this execution does not match the expectation (if the `node` process is not the Node.js executable and its `--version` parameter triggers a `rm -rf /`).
+Executing such code should be done conservatively on well-identified workloads (`python`, `node`, etc.) if they provide a way to get their versions without side-effect.
+
+##### Measurable Impact on Gathering Time & Payload
+
+The workload data is gathered every 12 hours by the Insights Operator.
+Gathering the runtime data will increase the overall executing time of the workload gatherer. The increase must remain reasonable.
+
+In the existing initial implementation of the insights-runtime-extractor, it takes __~200 ms to scan a single container__.
+As the inspection is done in parallel on each worker node, its overall duration is impacted by
+the density of the cluster (number of containers per worker node).
+On a fresh OpenShift cluster (with no user workload and only OpenShift own containers), it takes ~15 seconds to scan all containers.
+For a 8-worker node with 800 containers, assuming the containers are spread evenly, it would take around 20 seconds to scan the whole cluster. In practice, it will be a bit longer and capped by the inspection of the worker node with the most containers.
+
+The additional data in the `workloadRuntimeInfoContainer` will increase the size of the insights payload that is archived and sent to Red Hat backends.
+
+The additional data adds __~200 uncompressed bytes per WorkloadContainerShape in the JSON payload__.
+
+This additional payload does not scale with the number of pods but with the number of `WorkloadContainerShape` which is difficult to estimate. There is an upper limit of 8000 pods
+in the Insights operator but that does not give a limit on the number of containers. It is also very likely that some of the pods are replicas and their containers would only count for 1 `WorkloadContainerShape` in their gathered data.
+
+#### UX Risk & Mitigation
+
+The additional runtime info added to the gathered data will be exposed to the user though the archives generated by the Insights operator. As the values of the field are hashed, the user will not be able to determine their corresponding content.
+
+Exposing these data to the user with a clean user interface (eg within console.redhat.com) is out of scope of this proposal.
+
+### Drawbacks
+
+If implemented, Red Hat will document the addition of this new collection. We  introduce the risk that some number of customers who would have or are sending telemetry data via the Insights operator will choose to turn it off because they are uncomfortable sharing this additional data. 
```

Comment:
Is there a knob to just control whether this data is collected or not?

Reply:
No. I added a note about this as something that could be addressed in the `GA` stage by adding a flag in the openshift insights operator configuration.